draft-robert-privacypass-batched-tokens.md

```diff
@@ -234,30 +241,30 @@ def FinalizeBatch(input, blind, evaluatedElements, blindedElements, proof):
                  evaluatedElements, proof) == false:
     raise VerifyError
 
-  outputs = []
+  output = nil
   for evaluatedElement in evaluatedElements:
     N = G.ScalarInverse(blind) * evaluatedElement
     unblindedElement = G.SerializeElement(N)
     hashInput = I2OSP(len(input), 2) || input ||
                 I2OSP(len(unblindedElement), 2) || unblindedElement ||
                 "Finalize"
-    outputs.append(Hash(hashInput))
+    output = concat(output, Hash(hashInput))
 
-  return outputs
+  return output
 ~~~
 
-If this succeeds, the Client then constructs a Token as follows, where
-`authenticator` is the element of `autheticator_values` that corresponds to the
-nonce was sampled in {{client-to-issuer-request}} and that the client wants to
-redeem:
+If this succeeds, the Client then constructs `Nr` Token values as follows, where
+`authenticator_i` is the i-th element of `authenticator_values` that corresponds
+to the i-th nonce was sampled in {{client-to-issuer-request}} and that the client
+wants to redeem:
 
 ~~~
 struct {
     uint16_t token_type = 0x0003
-    uint8_t nonce[32];
+    uint8_t nonce_i[32];
```

Comment:
Just a nit, but aren't the tokens supposed to be exactly identical in their structure for all issuance protocols? I know it's just the name of the fields, but technically it makes the struct different.

Reply:
Fair point! I reverted this.